sas9.4-viya/azuredeploy.json

```diff
@@ -0,0 +1,3683 @@
+                        "createOption": "FromImage",
+                        "caching": "ReadWrite",
+                        "managedDisk": {
+                            "storageAccountType": "Premium_LRS"
+                        },
+                        "diskSizeGB": 64
+                    },
+                    "dataDisks": [
+                        {
+                            "name": "[concat(variables('ansible_vm_name'),'-playbook')]",
+                            "diskSizeGB": 50,
+                            "lun": 0,
+                            "createOption": "Empty"
+                        }
+                    ]
+                },
+                "osProfile": {
+                    "computerName": "[variables('ansible_vm_name')]",
+                    "adminUsername": "[variables('primary_user_name')]",
+                    "linuxConfiguration": {
+                        "disablePasswordAuthentication": true,
+                        "ssh": {
+                            "publicKeys": [
+                                {
+                                    "path": "[concat('/home/', variables('primary_user_name'), '/.ssh/authorized_keys')]",
+                                    "keyData": "[parameters('SSHPublicKey')]"
+                                }
+                            ]
+                        }
+                    }
+                },
+                "networkProfile": {
+                    "networkInterfaces": [
+                        {
+                            "id": "[resourceId('Microsoft.Network/networkInterfaces', variables('ansible_nw_interface'))]"
+                        }
+                    ]
+                },
+                "diagnosticsProfile": {
+                    "bootDiagnostics": {
+                        "enabled": true,
+                        "storageUri": "[reference(variables('diagnostic_storagegroup_name')).primaryEndpoints.blob]"
+                    }
+                },
+                "proximityPlacementGroup": {
+                    "id": "[resourceId('Microsoft.Compute/proximityPlacementGroups',variables('ppg_name'))]"
+                }
+            },
+            "resources": []
+        },
+        {
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2019-10-01",
+            "name": "AnsibleRoleAssignment",
+            "dependsOn": [
+                "[resourceId('Microsoft.Compute/virtualMachines', variables('ansible_vm_name'))]",
+                "[resourceId('Microsoft.Resources/deployments','AnsibleAccessPolicy')]"
+            ],
+            "properties": {
+                "mode": "Incremental",
+                "templateLink": {
+                    "uri": "[variables('role_assignment_template')]"
+                },
+                "parameters": {
+                    "roleAssignmentName": {
+                        "value": "[guid(variables('ansible_vm_name'),resourceGroup().id)]"

```

Comment:
the seed for the guid needs to include the prinicipal, scope and role

Reply:
done